src/helpers.ts

```diff
@@ -0,0 +1,126 @@
+import * as Ably from 'ably';
+
+import { MessageEvents, RoomReactionEvents } from './events.js';
+import { Message } from './Message.js';
+import { parseMessage } from './messageParser.js';
+import { Reaction } from './Reaction.js';
+import { parseReaction } from './reactionParser.js';
+
+/**
+ * Enum for chat entity types. This is used to determine the type of entity encoded in an inboundMessage-like object, and
+ * subsequently which method should be used to parse the entity.
+ */
+export enum ChatEntityType {
+  /**
+   * Represents a chat message type.
+   */
+  ChatMessage = 'chatMessage',
+  /**
+   * Represents a reaction type.
+   */
+  Reaction = 'reaction',
+}
+
+/**
+ * Helper function to validate the encoded object.
+ *
+ * @param encoded - The object to validate.
+ * @throws {@link ErrorInfo} - If the encoded object is invalid.
+ */
+function validateEncoded(encoded: unknown): void {
+  if (typeof encoded !== 'object' || encoded === null) {
```

Comment:
Do we need the `encoded === null` part here? Won't this case just be caught by the first condition?

Reply:
Sorry, I don't quite understand, probably my ts knowledge is lacking here, but the `typeof null === 'object'` is true, so doesn't that mean encoded can be null and pass the check if we don't check specifically for null? :)

Reply:
Well oh my... JavaScript never fails to surprise. Ignore this!